batch/batch/server/server.py

```diff
@@ -884,11 +999,13 @@ def __anext__(self):
         pod_name = pod.metadata.name
         seen_pods.add(pod_name)
 
-        job = pod_name_job.get(pod_name)
+        job = await Job.from_record(await db.jobs.get_record_by_pod(pod_name))
         if job and not job.is_complete():
             await update_job_with_pod(job, pod)
 
-    for pod_name, job in pod_name_job.items():
+    pod_jobs = [await Job.from_record(record) for record in await db.jobs.get_records_where({'pod_name': 'NOT NULL'})]
+    for job in pod_jobs:
```

Comment:
General points:
1. we should almost always avoid iterating a list of awaitables because it reduces concurrency.
2. when order does not matter we should use `asyncio.as_completed` which returns tasks in the order the complete

```python3
future_jobs = await db.jobs.get_records_where({'pod_name': 'NOT NULL'})
async for job in asyncio.as_completed(*[Job.from_record(record) for record in future_jobs]):
    ...
```

But, as noted above, I think `from_record` should not be `async` anyway.

Reply:
Good point. This is in other places. Again, future PR. Let's get this one in.